packages/astro/src/core/build/vite-plugin-analyzer.ts

```diff
@@ -55,13 +89,26 @@ export function vitePluginAnalyzer(internals: BuildInternals): VitePlugin {
 					}
 					internals.discoveredScripts.add(moduleId);
 
+					// TODO: map raw URL (?type=script) to client build ready URL
+					// This will inject the raw URL as a script tag as-is,
+					// which will fail to map with the client build output.
+					pageData.propagatedScripts = propagatedMapByImporter;
```

Comment:
This is the part left to figure out. Today, this naively passes the unformatted URL as a client build entrypoint. This gets mangled by Vite in the output (`script.astro?type=script` -> `script.astro.fancytype.2j234k2.js`).